src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java

```diff
@@ -0,0 +1,231 @@
+package cz.muni.pa165.teamwhite.formula1.Dao;
+
+
+import cz.muni.pa165.teamwhite.formula1.PersistenceSampleApplicationContext;
+import cz.muni.pa165.teamwhite.formula1.dao.CarDao;
+import cz.muni.pa165.teamwhite.formula1.dao.ComponentDao;
+import cz.muni.pa165.teamwhite.formula1.dao.DriverDao;
+import cz.muni.pa165.teamwhite.formula1.entity.Car;
+
+import javax.transaction.Transactional;
+import javax.validation.ConstraintViolationException;
+
+import cz.muni.pa165.teamwhite.formula1.entity.Component;
+import cz.muni.pa165.teamwhite.formula1.entity.Driver;
+import cz.muni.pa165.teamwhite.formula1.enums.ComponentType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceUnit;
+import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author Tomas Sedlacek
+ */
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+@Transactional
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+public class CarTests extends AbstractTestNGSpringContextTests {
+    @PersistenceUnit
+    private EntityManagerFactory emf;
+
+    @Autowired
+    private DriverDao driverDao;
+
+    @Autowired
+    private ComponentDao componentDao;
+
+    @Autowired
+    private CarDao carDao;
+
+    @Test(expectedExceptions = ConstraintViolationException.class)
+    public void CarWithNullNameTest() {
+        Car car = new Car(null, null, null);
+
+        carDao.create(car);
+    }
+
+    @Test
+    @Rollback(value = true)
```

Comment:
Rollback by tu nemusel asi byt, zkus to bez nej :smile:

Reply:
pushnul jsem to driv, nez jsem nasel to vlakno na stackoverflow :smiley: